### PR TITLE
Bug 1798697: Show support links

### PR DESCRIPTION
**Pull request by rebeccaalpert**
Support links were removed at the last minute from the original OLM workflows PR. Re-instantiating work.

Dev Catalog:
Screen Shot 2020-02-06 at 2 46 44 PM

OperatorHub:
Screen Shot 2020-02-06 at 2 47 26 PM

Installed Operators:
Screen Shot 2020-02-06 at 2 47 47 PM

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1798697.

Heads up @openshift/team-ux-review.

### CODE REVIEW

File: frontend/public/components/catalog/catalog-item-details.jsx [hunk 1]
```diff
@@ -41,6 +41,9 @@ export class CatalogTileDetails extends React.Component {
       documentationUrl,
       sampleRepo,
     } = this.props.item;
+    const {
+      'marketplace.openshift.io/support-workflow': marketplaceSupportWorkflow,
+    } = this.props.item.obj.csv.metadata.annotations;
```
Comment (spadgett) [inline frontend/public/components/catalog/catalog-item-details.jsx]: We should just set this as `supportUrl` when the catalog item is created

File: frontend/public/components/catalog/catalog-item-details.jsx [hunk 2]
```diff
@@ -67,7 +70,18 @@ export class CatalogTileDetails extends React.Component {
...
                 {tileProvider && <PropertyItem label="Provider" value={tileProvider} />}
-                {supportUrl && <PropertyItem label="Support" value={supportUrlLink} />}
+                {(supportUrl || marketplaceSupportWorkflow) && (
+                  <PropertyItem
+                    label="Support"
+                    value={
+                      marketplaceSupportWorkflow ? (
+                        <ExternalLink href={marketplaceSupportWorkflow} text="Get Support" />
+                      ) : (
+                        supportUrlLink
```
Comment (spadgett) [inline frontend/public/components/catalog/catalog-item-details.jsx]: I'd think we'd want to render this as a link regardless of how it was set.

Comment (rebeccaalpert) [reply frontend/public/components/catalog/catalog-item-details.jsx]: It is; supportUrlLink is defined as a link further up. I moved marketplaceSupportWorkflow up there in the latest commit if that works.

**Comment by rebeccaalpert** [comment]
```
error: build error: no such image
2020/02/06 21:05:37 No custom metadata found and prow metadata already exists. Not updating the metadata.
2020/02/06 21:05:38 Ran for 8m22s
2020/02/06 21:05:39 Submitted failure event to sentry (id=7ba5e9ca9cb54149af0627c0ecc4ad55)
error: could not run steps: step console failed: could not wait for build: the build console failed after 7m25s with reason DockerBuildFailed: Docker build strategy has failed.
```

/retest

File: frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx [hunk 3]
```diff
@@ -624,6 +624,9 @@ export const ClusterServiceVersionDetails: React.SFC<ClusterServiceVersionDetail
   props,
 ) => {
   const { spec, metadata, status } = props.obj;
+  const {
+    'marketplace.openshift.io/support-workflow': marketplaceSupportWorkflow,
+  } = props.obj.metadata.annotations;
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx]:
```suggestion
  } = metadata.annotations;
```

File: frontend/public/components/catalog/catalog-item-details.jsx (hunk 1 above)
Comment (spadgett) [reply frontend/public/components/catalog/catalog-item-details.jsx]: This should be set as `supportUrl` here:

https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/dev-catalog.ts#L70

File: frontend/public/components/catalog/catalog-item-details.jsx [hunk 4]
```diff
@@ -41,11 +41,16 @@ export class CatalogTileDetails extends React.Component {
...
+    const {
+      'marketplace.openshift.io/support-workflow': marketplaceSupportWorkflow,
+    } = this.props.item.obj.csv.metadata.annotations;
     const { plans } = this.state;
 
     const creationTimestamp = _.get(obj, 'metadata.creationTimestamp');
 
-    const supportUrlLink = <ExternalLink href={supportUrl} text="Get support" />;
+    const supportUrlLink = (
+      <ExternalLink href={supportUrl || marketplaceSupportWorkflow} text="Get Support" />
```
Comment (spadgett) [inline frontend/public/components/catalog/catalog-item-details.jsx]: This change won't be needed if you set `item.supportUrl`

**Comment by rebeccaalpert** [comment]
Addressed feedback.

File: frontend/public/components/catalog/catalog-item-details.jsx (hunk 1 above)
Comment (rebeccaalpert) [reply frontend/public/components/catalog/catalog-item-details.jsx]: Thanks!

File: frontend/packages/operator-lifecycle-manager/src/dev-catalog.ts [hunk 5]
```diff
@@ -67,7 +67,7 @@ export const normalizeClusterServiceVersions = (
       href: `/ns/${desc.csv.metadata.namespace}/clusterserviceversions/${
         desc.csv.metadata.name
       }/${referenceForProvidedAPI(desc)}/~new`,
-      supportUrl: null,
+      supportUrl: _.get(desc, ['csv', 'metadata', 'annotations', 'marketplace.openshift.io/support-workflow'], ''),
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/dev-catalog.ts]: Prefer optional chaining

```suggestion
      supportUrl: desc.csv.metadata.annotations?.['marketplace.openshift.io/support-workflow'],
```

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining

File: frontend/public/components/catalog/catalog-item-details.jsx [hunk 6]
```diff
@@ -45,7 +45,9 @@ export class CatalogTileDetails extends React.Component {
 
     const creationTimestamp = _.get(obj, 'metadata.creationTimestamp');
 
-    const supportUrlLink = <ExternalLink href={supportUrl} text="Get support" />;
+    const supportUrlLink = (
+      <ExternalLink href={supportUrl} text="Get Support" />
```
Comment (spadgett) [inline frontend/public/components/catalog/catalog-item-details.jsx]: I'd revert this change. There shouldn't need to be any changes in this file now that we're passing the right `supportUrl`

Comment (rebeccaalpert) [reply frontend/public/components/catalog/catalog-item-details.jsx]: I'll remove the linter parenthesis, but I'd argue "Get Support" should be the same case throughout (it would be "Get Support" everywhere but this file if I revert everything).

Comment (spadgett) [reply frontend/public/components/catalog/catalog-item-details.jsx]: Everywhere else it's `Get support`, so I'd recommend changing the new link to match instead of changing this one. That's more inline with the Patternfly recommendations and other similar links in console like `View events`, `View alerts`, etc.

https://github.com/openshift/console/search?q=%22get+support%22&unscoped_q=%22get+support%22

File: frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx [hunk 7]
```diff
@@ -652,6 +655,14 @@ export const ClusterServiceVersionDetails: React.SFC<ClusterServiceVersionDetail
                 <dd>
                   {spec.provider && spec.provider.name ? spec.provider.name : 'Not available'}
                 </dd>
+                {marketplaceSupportWorkflow && (
+                  <>
+                    <dt>Support</dt>
+                    <dd>
+                      <ExternalLink href={marketplaceSupportWorkflow} text="Get Support" />
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx]:
```suggestion
                      <ExternalLink href={marketplaceSupportWorkflow} text="Get support" />
```

File: frontend/public/components/catalog/catalog-item-details.jsx [hunk 8]
```diff
@@ -45,7 +45,7 @@ export class CatalogTileDetails extends React.Component {
 
     const creationTimestamp = _.get(obj, 'metadata.creationTimestamp');
 
-    const supportUrlLink = <ExternalLink href={supportUrl} text="Get support" />;
+    const supportUrlLink = <ExternalLink href={supportUrl} text="Get Support" />;
```
Comment (spadgett) [inline frontend/public/components/catalog/catalog-item-details.jsx]:
```suggestion
    const supportUrlLink = <ExternalLink href={supportUrl} text="Get support" />;
```

**Comment by rebeccaalpert** [comment]
All set.
